grr/server/grr_response_server/bin/grrafana.py

```diff
@@ -98,115 +214,38 @@ def _OnRoot(self, request: JSONRequest) -> JSONResponse:
-    if "type" in request.json:
-      # Grafana request issued on Panel > Queries page. Grafana expects the
-      # list of metrics that can be listed on the dropdown-menu
-      # called "Metric".
-      response = AVAILABLE_METRICS
-    else:
-      # Grafana request issued on Variables > New/Edit page for variables of
-      # type query. Grafana expectes the list of possible values of
-      # the variable. At the moment, GRRafana doesn't support such variables,
-      # so it returns no possible values.
-      response = []
+    response = list(AVAILABLE_METRICS_DICT.keys())
     return JSONResponse(response=response)
 
   def _OnQuery(self, request: JSONRequest) -> JSONResponse:
     """Retrieves datapoints for Grafana.
-    
+
     Given a client ID as a Grafana variable and targets (resource usages),
     returns datapoints in a format Grafana can interpret."""
     json_data = request.json
-    requested_client_id = _ExtractClientIdFromVariable(
-        json_data)  # There must be a ClientID variable declated in Grafana.
     requested_targets = [entry["target"] for entry in json_data["targets"]]
-    list_targets_with_datapoints = _FetchDatapointsForTargets(
-        client_id=requested_client_id,
-        start_range=json_data["range"]["from"],
-        end_range=json_data["range"]["to"],
-        targets=requested_targets)
-    response = [t._asdict() for t in list_targets_with_datapoints]
+    targets_with_datapoints = [
+        AVAILABLE_METRICS_DICT[target].ProcessQuery(json_data)
+        for target in requested_targets
+    ]
+    response = [t._asdict() for t in targets_with_datapoints]
     return JSONResponse(response=response)
 
   def _OnAnnotations(self, request: JSONRequest) -> JSONResponse:
     pass
 
 
-_Datapoint = Tuple[float, int]
-_Datapoints = List[_Datapoint]
-_TargetWithDatapoints = collections.namedtuple("TargetWithDatapoints",
-                                              ["target", "datapoints"])
-
-
-def _FetchDatapointsForTargets(
-    client_id: str, start_range: str,
-    end_range: str,
-    targets: Iterable[str]) -> List[_TargetWithDatapoints]:
-  """Fetches a list of <datapoint, timestamp> tuples for each target 
-  metric from Fleetspeak database."""
-  start_range_timestamp = timeToProtoTimestamp(start_range)
-  end_range_timestamp = timeToProtoTimestamp(end_range)
-  records_list = fleetspeak_utils.FetchClientResourceUsageRecords(
-      client_id, start_range_timestamp, end_range_timestamp)
-  response = []
-  for target in targets:
-    datapoints_for_single_target = _CreateDatapointsForTarget(
-        target, records_list)
-    response.append(
-        _TargetWithDatapoints(target=target,
-                             datapoints=datapoints_for_single_target))
-  return response
-
-
 def timeToProtoTimestamp(
     grafana_time: str) -> timestamp_pb2.Timestamp:
-  date = dateutil.parser.parse(grafana_time)
+  date = dateutil.parser.parse(grafana_time)  # type: ignore
```

Comment:
All `type: ignore` should be accompanied by a comment (preferably linking to a particular bug).